core/core/src/main/java/androidx/core/app/ActivityCompat.java

```diff
@@ -752,6 +756,19 @@ private Api31Impl() {
         static boolean isLaunchedFromBubble(@NonNull final Activity activity)  {
             return activity.isLaunchedFromBubble();
         }
+
+        @DoNotInline
+        static boolean shouldShowRequestPermissionRationale(Activity activity, String permission) {
+            try {
+                // Fix shouldShowRequestPermissionRationale causing memory leak in Android 12
```

Comment:
Also add a comment why calling this via reflection  fixes the issue.

Reply:
> Add a link to the leak in question.
> 
> Which version was it fixed in?

Would you like to add it to the code comments or issue comments?

Reply:
Yes, please leave a comment with links to the bugs that we are trying to work around.

Reply:
> Add a link to the leak in question.
> 
> Which version was it fixed in?
>
> Also add a comment why calling this via reflection fixes the issue.

 1. Background of the problem：Fix shouldShowRequestPermissionRationale causing memory leak in Android 12，
    this problem has been fixed on the Android 12L system, but it is still a historical problem for the Android 12 system
 2. The reason for the problem: The culprit is that the PermissionUsageHelper holds the Context object as a field,
    and calls AppOpsManager.startWatchingStarted in the constructor to start the monitoring,
    so that the PermissionUsageHelper object will be added to the AppOpsManager#mStartedWatchers collection,
    which will lead to active calls in the Activity When finishing, stopWatchingStarted is not used to remove the watch,
    which causes the Activity object to be held in the AppOpsManager#mStartedWatchers collection,
    which indirectly causes the Activity object to not be recycled by the system.
 3. The solution to the problem: The handling of this problem is also very simple and rude, that is,
    to replace the Context parameter passed in the outer layer from the Activity object to the Application object,
    because the Application life cycle is relatively long, but there is only the shouldShowRequestPermissionRationale method in the Activity,
    and What if there is no such method in Application? Looking at the implementation of this method, in fact,
    that method will eventually call the PackageManager.shouldShowRequestPermissionRationale method (hidden API, but not in the blacklist),
    so as long as the PackageManager object can be obtained, and finally use reflection to execute this method , which avoids memory leaks.

Reply:
@liutikas Hello, issue comments and code comments for this piece have been added.